cli/src/main/java/com/liferay/blade/cli/Extensions.java

```diff
@@ -387,6 +391,9 @@ private void _extractBladeExtensions(Path extensionsDirectory) throws IOExceptio
 
 						commandsToRemove.add(baseCommand);
 					}
+					else if (!profileNames.isEmpty() && !profileNames.contains(profileName)) {
```

Comment:
How is this block in 394 different or also needed when we have line 400-407?  Seems like those could be combined somehow?